### PR TITLE
Add test files for methods and validation

### PR DESCRIPTION
**Pull request by devalibello**
This pull request adds RSpec test files for validations and methods in models

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:user) { User.create(name: 'John') }
+
+  it 'title valid with a title' do
+    first_user = User.first
+    post = Post.new(author: first_user, title: 'my first post', commentscounter: 0, likescounter: 0)
+    expect(post).to be_valid
+  end
```
Comment (OyePriscilla) [inline spec/models/post_spec.rb]: - [ ] This particular test case is failing, I got the error message screenshot below 👇 

Untitled

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  it 'updates the postscounter after creation' do
+    Post.create(
+      title: 'Sample Post',
+      commentscounter: 0,
+      likescounter: 0,
+      author: user
+    )
+    user.reload
+    expect(user.postcounter).to eq(1)
+  end
```
Comment (OyePriscilla) [inline spec/models/post_spec.rb:32]: - [ ] This test case is failing as well, I got the error message screenshot below 👇

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @devalibello 👏 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

[so-close](https://user-images.githubusercontent.com/103386183/196423863-0f522555-1b03-455e-a452-b822cdcfc398.gif)

## Highlights
Used validations for models. ✔️ 
Implemented unit tests for models. ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 2 above)
Comment (devalibello) [reply spec/models/post_spec.rb:32]: Hello @OyePriscilla 
I trust you are doing great.
Could you please cross-check the code and tests, everything works fine from my end 😄 .
image

If the error is from my code, I will kindly request you specify the faulty lines of code to me 🐛 

Thanks

Comment (OyePriscilla) [reply spec/models/post_spec.rb:32]: Ok @devalibello, let's have a zoom call.

Comment (OyePriscilla) [reply spec/models/post_spec.rb:32]: @devalibello, please kindly take note that a senior colleague also got the same error messages I pointed out. To resolve the issue, please ensure that all the needed parameters are put in to create a user in post_spec.rb file line 4.

Untitled

*example* `let(:user)**** { User.create(name: "me", photo: "me", bio: "me", postcounter: 0) }
`
Please don't forget to resolve this failing test as well 👇 

Untitled

Comment (devalibello) [reply spec/models/post_spec.rb:32]: @OyePriscilla Well noted, with thanks.
Thanks for your detailed feedback. I am available now if a call is okay with you 😄. There is always something to learn from others 👨‍🎓

**Review by mukhammadrizooff — APPROVED** [review]
Hi @devalibello, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions
- [ ] [OPTIONAL] You did a well, good job 👍🏼 

image

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**